runtime-components/non-repudiation-api/src/test/scala/com/daml/nonrepudiation/api/v1/SignedPayloadsEndpointSpec.scala

```diff
@@ -0,0 +1,49 @@
+// Copyright (c) 2021 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.nonrepudiation.api.v1
+
+import java.time.Instant
+
+import com.daml.nonrepudiation.{
+  AlgorithmString,
+  FingerprintBytes,
+  PayloadBytes,
+  SignatureBytes,
+  SignedPayload,
+}
+import org.scalatest.flatspec.AnyFlatSpec
+import org.scalatest.matchers.should.Matchers
+
+final class SignedPayloadsEndpointSpec extends AnyFlatSpec with Matchers {
+
+  behavior of "SignedPayloadEndpoint.toResponse"
+
+  it should "convert the backend signed payload into the expected representation" in {
+
+    val algorithm = "algorithm"
+    val fingerprint = "fingerprint".getBytes
+    val payload = "payload".getBytes
+    val signature = "signature".getBytes
+    val timestamp = 42L
+
+    val signedPayload =
+      SignedPayload(
+        algorithm = AlgorithmString.wrap(algorithm),
+        fingerprint = FingerprintBytes.wrap(fingerprint),
+        payload = PayloadBytes.wrap(payload),
+        signature = SignatureBytes.wrap(signature),
+        timestamp = Instant.ofEpochMilli(timestamp),
+      )
+
+    SignedPayloadsEndpoint.toResponse(signedPayload) shouldBe SignedPayloadsEndpoint.Response(
+      algorithm = algorithm,
+      fingerprint = "ZmluZ2VycHJpbnQ=",
```

Comment:
I assume that’s base64 encoded? maybe worth adding a comment to make the test a bit clearer (or just do the base64 encoding but then this test is pretty close to just replicating the logic which is also not super useful).

Reply:
Base64, URL-safe, I'll add a comment, thanks for pointing this out.

I preferred to not replicate the logic but rather have a "golden sample" of some sort. I was mostly interesting in making sure that this was not utterly broken.

Reply:
Addressed by https://github.com/digital-asset/daml/pull/8982/commits/cdd891222d7f92ba6a0da3aca5a3ab95a13f0045